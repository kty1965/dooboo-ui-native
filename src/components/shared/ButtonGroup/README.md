# ButtonGroup

[![Npm Version](http://img.shields.io/npm/v/@dooboo-ui/native-button-group.svg?style=flat-square)](https://npmjs.org/package/@dooboo-ui/native-button-group)
[![Downloads](http://img.shields.io/npm/dm/@dooboo-ui/native-button-group.svg?style=flat-square)](https://npmjs.org/package/@dooboo-ui/native-button-group)

> [ButtonGroup] is an `ios` section like component. You can have as may selectors as you want.

![ButtonGroup](https://user-images.githubusercontent.com/27461460/62305265-8c2a2600-b4ba-11e9-83df-af0ac2f4a3f6.gif)

## Props

|                   | necessary | types                  | default                  |
| ----------------- | --------- | ---------------------- | ------------------------ |
| testID            |           | string                 |                          |
| containerStyle    |           | `StyleProp<ViewStyle>` |                          |
| style             |           | `StyleProp<ViewStyle>` |                          |
| viewStyle         |           | `StyleProp<ViewStyle>` |                          |
| selectedViewStyle |           | `StyleProp<ViewStyle>` |                          |
| textStyle         |           | `StyleProp<TextStyle>` |                          |
| selectedTextStyle |           | `StyleProp<TextStyle>` |                          |
| data              |           | string[]               | ['Option 1', 'Option 2'] |
| onPress           |           | (i: number) => void    |                          |

## Installation

```sh
yarn add @dooboo-ui/native
```

or

```sh
yarn add @dooboo-ui/native-button-group
```

## Getting started

- Import

  ```javascript
  import { ButtonGroup } from '@dooboo-ui/native';
  // or
  import ButtonGroup from '@dooboo-ui/native-button-group';
  ```

- Usage

  ```javascript
  function Page(props: Props) {
    const data = ['Option 1', 'Option 2', 'Option 3', 'Option 4'];

    const [option, setOption] = useState('Option 1');

    const selectOption = (index: number) => {
      setOption(data[index]);
      switch (index) {
        case 0:
          setOption('Option 1');
          break;
        case 1:
          setOption('Option 2');
          break;
      }
    };

    return (
      <Container>
        <ButtonGroup
          testID="BTN_GROUP"
          style={{ marginTop: 40 }}
          onPress={(index: number) => selectOption(index)}
          data={data}
        />
        <View
          style={{
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <Text
            style={{
              fontSize: 32,
            }}
          >
            {option}
          </Text>
        </View>
      </Container>
    );
  }
  ```
