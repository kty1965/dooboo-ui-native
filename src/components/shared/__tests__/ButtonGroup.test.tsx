import 'react-native';

import * as React from 'react';

import { RenderResult, act, fireEvent, render } from '@testing-library/react-native';

import ButtonGroup from '../ButtonGroup';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

const createTestProps = (obj: object): object => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});

describe('[ButtonGroup] render', () => {
  beforeEach(() => {
    props = createTestProps({});
    component = <ButtonGroup {...props} />;
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON = renderer
      .create(component)
      .toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onClick', () => {
      const btn1 = testingLib.queryByTestId('CHILD_1');
      act(() => {
        fireEvent.press(btn1);
      });
    });
  });
});
