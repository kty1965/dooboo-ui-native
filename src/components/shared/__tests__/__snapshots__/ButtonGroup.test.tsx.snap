// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ButtonGroup] render renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "transparent",
        "borderColor": "rgb(62,126,255)",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 32,
        "justifyContent": "center",
        "marginTop": 24,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
    testID="CHILD_0"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgb(62,126,255)",
            "borderColor": "rgb(62,126,255)",
            "height": 30,
            "justifyContent": "center",
          },
          Object {
            "borderRightWidth": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 14,
            "textAlign": "center",
          }
        }
      >
        option 1
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
    testID="CHILD_1"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "white",
            "borderColor": "rgb(62,126,255)",
            "height": 30,
            "justifyContent": "center",
          },
          null,
        ]
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(62,126,255)",
            "fontSize": 14,
            "textAlign": "center",
          }
        }
      >
        option 2
      </Text>
    </View>
  </View>
</View>
`;
